pandas/core/frame.py

```diff
@@ -1440,7 +1456,7 @@ def to_numpy(
 
         return result
 
-    def to_dict(self, orient="dict", into=dict):
+    def to_dict(self, orient: str = "dict", into=dict) -> Union[Dict, List, Mapping]:
```

Comment:
Mapping should be enough here, why did you add List?

Reply:
You can at least get rid of Dict then - no reason to include Dict and Mapping unless I am overlooking something

Reply:
Yes, that's right - done

Reply:
> with `orient="records"` we return a list:

we should probably overload using Literal to avoid Union return types. but not in this pass.